### PR TITLE
[Vectorized][Feature] support some bitmap functions

### LINKED ISSUE #8137: [Vectorized][Feature] support some bitmap functions
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

bitmap_and_not_count/bitmap_or/bitmap_xor/bitmap_xor_count/bitmap_and/bitmap_and_not
bitmap_subset_in_range/bitmap_subset_limit/bitmap_and_count/sub_bitmap/bitmap_has_all/bitmap_or_count

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #8137 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_bitmap_variadic.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        auto& vec_res = col_res->get_data();
+        vec_res.resize(input_rows_count);
+
+        Impl::vector_vector(argument_columns, argument_size, input_rows_count, vec_res);
+        block.replace_by_position(result, std::move(col_res));
+        return Status::OK();
+    }
+};
+
+using FunctionBitmapOr = FunctionBitMapVariadic<BitmapOr>;
```
Comment (zbtzbtzbt) [inline be/src/vec/functions/function_bitmap_variadic.cpp:214]: 为啥不用原来的bitmap and/or/xor了？

Comment (zhangstar333) [reply be/src/vec/functions/function_bitmap_variadic.cpp:214]: now need support bitmap_or(bitmap, bitmap, ...);

File: be/src/vec/functions/function_bitmap_variadic.cpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+#include "vec/data_types/data_type.h"
+#include "vec/data_types/data_type_bitmap.h"
+#include "vec/functions/function_const.h"
+#include "vec/functions/function_string.h"
+#include "vec/functions/function_totype.h"
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris::vectorized {
+
+struct BitmapAnd {
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap_variadic.cpp:28]: use template or macro to simple the code

File: be/src/vec/functions/function_bitmap_variadic.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+#define BITMAP_FUNCTION_COUNT_VARIADIC(CLASS, FUNCTION_NAME, OP)                                   \
+    struct CLASS {                                                                                 \
+        static constexpr auto name = #FUNCTION_NAME;                                               \
+        using ResultDataType = DataTypeInt64;                                                      \
+        using TData = std::vector<BitmapValue>;                                                    \
+        using ResTData = typename ColumnVector<Int64>::Container;                                  \
+        static Status vector_vector(ColumnPtr argument_columns[], size_t col_size,                 \
+                                    size_t input_rows_count, ResTData& res) {                      \
+            TData vals(input_rows_count);                                                          \
+            vals = assert_cast<const ColumnBitmap*>(argument_columns[0].get())->get_data();        \
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap_variadic.cpp:59]: TData vals

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**